### PR TITLE
Fix #1918 Show only a snackbar when space is insufficient

### LINKED ISSUE #1918: Toast and snackbar shown together when downloading a large zim with insufficient memory
**Describe the bug**
A toast and a snackbar are both shown when user tries to download a large zim and there's not enough storage. 

**Expected behavior**
Either a toast or a snackbar should be shown. Here, showing just a toast is not an option since the snackbar has an action. So showing just a snackbar with the required message is enough. 

**Steps to reproduce the behavior:**
1. Go to online section of the library
2. Try downloading a zim that will not fit in the internal storage of the phone
3. A toast, as well as a snackbar, is displayed

**Environment**
- Version of Kiwix Android : 3.2.1
- Device : Redmi K20 
- OS version : Android 10

**Screenshot**

screencap

**Additional Context**
I would like to work on this issue.

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #1918 

Now just a snackbar is shown with the appropriate message when storage is insufficient for downloading a zim instead of showing a toast and a snackbar both. 

**Screenshots** 

screencap

### CODE REVIEW

**Comment by macgills** [comment]
I don't think it is obvious what `Storage` means here as a button/action. "Switch/Change/Select Storage" might work better?

**Comment by yashk2000** [comment]
@macgills Such a message would occupy too much space. And clicking the button just opens a dialog that shows `available/total storage`. It doesn't allow users to free up memory or open the file explorer. I think that since it's not opening any major action, this title is fine. 
But the main reason is the space constraint. If we increase the size, the rest of the content will not be visible. 

Or maybe we could use a dialog box?
Please let me know of your views on this.

**Comment by macgills** [comment]
Show me what it looks like with "Change Storage"

**Comment by yashk2000** [comment]
@macgills it hides a major portion of the snackbar message.

screencap

**Comment by macgills** [comment]
image
https://material.io/components/snackbars/#anatomy explore the api and see if there is a way to do this, don't know off the top of my head

**Comment by yashk2000** [comment]
@macgills please review.

screencap

File: core/src/main/res/values/themes.xml [hunk 1]
```diff
@@ -16,6 +16,9 @@
     <item name="colorSecondary">?colorPrimary</item>
     <item name="colorSecondaryVariant">?colorPrimary</item>
 
+    <!--Material snackbar attributes-->
+    <item name="maxActionInlineWidth" tools:ignore="PrivateResource">@dimen/design_snackbar_action_inline_max_width</item>
```
Comment (macgills) [inline core/src/main/res/values/themes.xml]: Don't use a private resource

Comment (yashk2000) [reply core/src/main/res/values/themes.xml]: @macgills fixed this.

File: core/src/main/res/values/themes.xml [hunk 2]
```diff
@@ -17,7 +17,7 @@
     <item name="colorSecondaryVariant">?colorPrimary</item>
 
     <!--Material snackbar attributes-->
-    <item name="maxActionInlineWidth" tools:ignore="PrivateResource">@dimen/design_snackbar_action_inline_max_width</item>
+    <item name="maxActionInlineWidth">@dimen/snackbar_action_inline_max_width</item>
```
Comment (macgills) [inline core/src/main/res/values/themes.xml]: It doesn't need to be a dimen either, you can put 128dp in here, I highly doubt we will ever need to reuse this

Comment (yashk2000) [reply core/src/main/res/values/themes.xml]: @macgills done.

File: core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt [hunk 3]
```diff
@@ -37,6 +37,20 @@ fun View.snack(
   }.show()
 }
 
+fun View.snack(
+  string: String,
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt]: string is not a great parameter name, i should match the snackbar api which I would guess is `message`

Comment (yashk2000) [reply core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt]: @macgills renamed the variable.

**Comment by yashk2000** [comment]
@macgills is this pr fine now?

**Comment by macgills** [comment]
All good! Just absentmindedly forgot to merge.

**Pull request merged**